### ISSUE TITLE
With update to version 1.0.1, automatic reconnect attempts fail

### ISSUE THREAD

**Issue by iprado78** [stomp-js/rx-stomp#277]
My team is seeing an issue with the automatic reconnect behavior since updating to `1.0.1` (behavior does not exist when using  `0.3.5`).  Not sure if the issue is with the latest version of `stompjs` itself or with the Rx wrapper's integration with it.

If we connect our client, bring the server/broker down and back up, the reconnection attempts fail.  We see 

```
WebSocket connection to 'ws://localhost/ws' failed: WebSocket is closed before the connection is established.  

stomp-handler.ts:326 
```

Only client configuration we are sending on initiation is `brokeURL` and `reconnectDelay`. 

Full stack dump:

```
_closeWebsocket @ stomp-handler.ts:326
_closeOrDiscardWebsocket @ stomp-handler.ts:309
forceDisconnect @ stomp-handler.ts:319
forceDisconnect @ client.ts:582
(anonymous) @ client.ts:425
setTimeout (async)
(anonymous) @ client.ts:419
fulfilled @ byte.ts:8
Promise.then (async)
step @ byte.ts:8
(anonymous) @ byte.ts:8
push../node_modules/@stomp/stompjs/esm6/client.js.__awaiter @ byte.ts:8
_connect @ client.ts:402
(anonymous) @ client.ts:524
setTimeout (async)
_schedule_reconnect @ client.ts:523
onWebSocketClose @ client.ts:484
_onclose @ stomp-handler.ts:164
_closeWebsocket @ stomp-handler.ts:326
_closeOrDiscardWebsocket @ stomp-handler.ts:309
forceDisconnect @ stomp-handler.ts:319
forceDisconnect @ client.ts:582
(anonymous) @ client.ts:425
setTimeout (async)
(anonymous) @ client.ts:419
fulfilled @ byte.ts:8
Promise.then (async)
step @ byte.ts:8
(anonymous) @ byte.ts:8
push../node_modules/@stomp/stompjs/esm6/client.js.__awaiter @ byte.ts:8
_connect @ client.ts:402
(anonymous) @ client.ts:524
setTimeout (async)
_schedule_reconnect @ client.ts:523
onWebSocketClose @ client.ts:484
_onclose @ stomp-handler.ts:164
_closeWebsocket @ stomp-handler.ts:326
_closeOrDiscardWebsocket @ stomp-handler.ts:309
forceDisconnect @ stomp-handler.ts:319
forceDisconnect @ client.ts:582
(anonymous) @ client.ts:425
setTimeout (async)
(anonymous) @ client.ts:419
fulfilled @ byte.ts:8
Promise.then (async)
step @ byte.ts:8
(anonymous) @ byte.ts:8
push../node_modules/@stomp/stompjs/esm6/client.js.__awaiter @ byte.ts:8
_connect @ client.ts:402
(anonymous) @ client.ts:524
setTimeout (async)
_schedule_reconnect @ client.ts:523
onWebSocketClose @ client.ts:484
_onclose @ stomp-handler.ts:164
_closeWebsocket @ stomp-handler.ts:326
_closeOrDiscardWebsocket @ stomp-handler.ts:309
forceDisconnect @ stomp-handler.ts:319
forceDisconnect @ client.ts:582
(anonymous) @ client.ts:425
setTimeout (async)
(anonymous) @ client.ts:419
fulfilled @ byte.ts:8
Promise.then (async)
step @ byte.ts:8
(anonymous) @ byte.ts:8
push../node_modules/@stomp/stompjs/esm6/client.js.__awaiter @ byte.ts:8
_connect @ client.ts:402
(anonymous) @ client.ts:524
setTimeout (async)
_schedule_reconnect @ client.ts:523
onWebSocketClose @ client.ts:484
_onclose @ stomp-handler.ts:164
_closeWebsocket @ stomp-handler.ts:326
_closeOrDiscardWebsocket @ stomp-handler.ts:309
forceDisconnect @ stomp-handler.ts:319
forceDisconnect @ client.ts:582
(anonymous) @ client.ts:425
setTimeout (async)
(anonymous) @ client.ts:419
fulfilled @ byte.ts:8
Promise.then (async)
step @ byte.ts:8
(anonymous) @ byte.ts:8
push../node_modules/@stomp/stompjs/esm6/client.js.__awaiter @ byte.ts:8
_connect @ client.ts:402
(anonymous) @ client.ts:524
setTimeout (async)
_schedule_reconnect @ client.ts:523
onWebSocketClose @ client.ts:484
_onclose @ stomp-handler.ts:164
```

**Comment by kum-deepak** [comment]
A couple of new options were added in this set of releases. One of these is https://stomp-js.github.io/guide/stompjs/rx-stomp/upgrading-to-stompjs-6-rx-stomp-1.html#connectiontimeout with a default of 10,000ms. From the stack trace, it appears that it is hitting timeout during connect attempt and force-disconnects. Please try setting `connectiontimeout` to `0` to disable this feature.

Let me know if it helps.

**Comment by iprado78** [comment]
Thanks, that did resolve it

**Comment by kum-deepak** [comment]
I saw a brief demo of the application on the browser. I must say I am impressed. A beautiful showcase of all the technologies in the play. Please convey my regards to the team.

Not many use `rx-stomp`, even lesser who use `rx-stomp rpc`. I believe that you have been using this library for `rpc` as well as `stream`. Please let me know if anyone from your team would be willing to write an article (maybe as a tutorial) on this topic.

**Comment by iprado78** [comment]
Thank you and, yes, we use `rpc`.  I am new to the project but using `rpc` in the architecture goes back to before we started using `rx-stomp`.  Let me try to get some context on the architectural choice from someone who's been around (may be next week as a lot of people are off), and then I could write something in January about why we chose `rpc` and then how we use the library for it.

**Comment by kum-deepak** [comment]
Many thanks, @iprado78. That will indeed be great. :+1:

**Comment by kum-deepak** [comment]
Released @stomp/stompjs@6.1.0 and @stomp/rx-stomp@1.0.2. It sets https://stomp-js.github.io/api-docs/latest/classes/Client.html#connectionTimeout to 0 to de-activate this feature.
